lib/validate.js

```diff
@@ -712,6 +712,32 @@ export const nwcSchema = object({
     })
 })
 
+export const blinkSchema = object({
+  apiKey: string()
+    .required('required')
+    .test(async (nwcUrl, context) => {
+      try {
+        await string().required('required').validate(nwcUrl)
+        await string().matches(/^blink_[A-Za-z0-9]+$/, { message: 'must match pattern blink_A-Za-z0-9' }).validate(nwcUrl)
+      } catch (err) {
+        return context.createError({ message: err.message })
+      }
+      return true
+    }),
+  currency: string()
+    .transform(value => value.toUpperCase())
+    .default('BTC')
+    .test(async (currency, context) => {
+      try {
+        const supportedCurrencies = ['USD', 'BTC']
+        await string().required('required').oneOf(supportedCurrencies, 'must be one of ' + supportedCurrencies.join(', ')).validate(currency)
+      } catch (err) {
+        return context.createError({ message: err.message })
+      }
+      return true
+    })
+})
```

Comment:
You don't need `.test` if you don't do custom validation. For NWC, I had to use it to parse the NWC url and individually validate sections of it + control validation order. You don't need that here.

Therefore, this should suffice:

```suggestion
export const blinkSchema = object({
  apiKey: string()
    .required('required')
    .matches(/^blink_[A-Za-z0-9]+$/, { message: 'must match pattern blink_A-Za-z0-9' }),
  currency: string()
    .transform(value => value.toUpperCase())
    .required('required')
    .default('BTC')
    .oneOf(['USD', 'BTC'], 'must be BTC or USD')
})
```

Reply:
done